File: api/calculate_winner_test.py
from calculate_winner import calculateWinner

def test_calculate_winner_no_winner():
    squares = [None] * 9
    result = calculateWinner(squares)["winner"]
    assert result == None

def test_calculate_winner_x_horizontally():
    squares = [
        "X", "X", "X",
        None, None, None,
        None, None, None
    ]
    result = calculateWinner(squares)["winner"]
    assert result == "X"

def test_calculate_winner_o_diagonally():
    squares = [
        "O", None, None,
        None, "O", None,
        None, None, "O"
    ]
    result = calculateWinner(squares)["winner"]
    assert result == "O"

def test_calculate_winner_x_vertically():
    squares = [
        "X", None, None,
        "X", None, None,
        "X", None, None
    ]
    result = calculateWinner(squares)["winner"]
    assert result == "X"

def test_calculate_winner_game_tie():
    squares = [
        "X", "O", "X",
        "O", "X", "O",
        "O", "X", "O"
    ]
    result = calculateWinner(squares)["winner"]
    assert result == None

File: front/__tests__/calculate_winner.test.mjs
/* eslint-disable jest/valid-expect */
/* eslint-disable no-unused-expressions */
import calculateWinner from "../src/models/calculate_winner.mjs";
import { expect } from 'chai';


describe("calculateWinner", () => {
    it("should return null if no winner", function() {
      const squares = Array(9).fill(null);
      const result = calculateWinner(squares);
      expect(result).to.be.null;
    });
  
    it("should return X if X wins horizontally", () => {
      const squares = [
        "X", "X", "X",
        null, null, null,
        null, null, null
    ];
      const result = calculateWinner(squares);
      expect(result).to.equal("X");
    });
  
    it("should return O if O wins diagonally", () => {
      const squares = [
        "O", null, null,
        null, "O", null,
        null, null, "O"
    ];
      const result = calculateWinner(squares);
      expect(result).to.equal("O");
    });
  
    it("should return X if X wins vertically", () => {
      const squares = [
        "X", null, null,
        "X", null, null,
        "X", null, null
    ];
      const result = calculateWinner(squares);
      expect(result).to.equal("X");
    });
  
    it("should return null if game is a tie", () => {
      const squares = [
        "X", "O", "X",
        "O", "X", "O",
        "O", "X", "O"
    ];
      const result = calculateWinner(squares);
      expect(result).to.be.null;
    });
});